src/test/kotlin/org/rust/ide/intentions/InjectLanguageIntentionTest.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.injected.editor.EditorWindow
+import com.intellij.lang.injection.InjectedLanguageManager
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiLanguageInjectionHost
+import com.intellij.psi.injection.Injectable
+import com.intellij.psi.util.PsiTreeUtil
+import org.intellij.lang.annotations.Language
+import org.intellij.plugins.intelliLang.inject.InjectLanguageAction
+import org.intellij.plugins.intelliLang.inject.InjectedLanguage
+
+class InjectLanguageIntentionTest : RsIntentionTestBase(InjectLanguageAction()) {
+    private fun checkAvailable(@Language("Rust") code: String) {
+        InlineFile(code.trimIndent()).withCaret()
+        check(intention.isAvailable(project, myFixture.editor, myFixture.file)) {
+            "Intention is not available"
+        }
+    }
+
+    private fun doTest(lang: String, @Language("Rust") code: String) {
+        InlineFile(code.trimIndent()).withCaret()
+        val language = InjectedLanguage.findLanguageById(lang)
+        val injectable = Injectable.fromLanguage(language)
+        InjectLanguageAction.invokeImpl(project, myFixture.editor, myFixture.file, injectable)
+        val host = findInjectionHost(myFixture.editor, myFixture.file)
+        check(host != null)
+        val injectedList = InjectedLanguageManager.getInstance(project).getInjectedPsiFiles(host!!)
+        check(injectedList != null && injectedList.size == 1)
+        val injectedPsi = injectedList!!.first().first
+        check(injectedPsi.language == language)
```

Comment:
Maybe use `assertEquals(language, injectedPsi.language)`? - it should produce more pretty error message